File: quickjs/test/polytraj7.js
import * as a from "liba.so";
var ctx = new a.polytraj7(0, 1, 0, 0)
ctx.gen(0, 10, 0, 10, 0, 0)
console.log(ctx.q)
console.log(ctx.v)
console.log(ctx.a)
console.log(ctx.j)
for (var i = 0; i < 10; i += 0.1) {
    console.log(i, ctx.pos(i), ctx.vel(i), ctx.acc(i), ctx.jer(0.5))
}

File: java/test/_polytraj7.java
import java.util.Arrays;
import liba.polytraj7;

class _polytraj7 {
    public static void main(String[] args) {
        polytraj7 obj = new polytraj7(0, 1, 0, 1, 0, 1, 0, 1, 0, 1);
        System.out.println(Arrays.toString(obj.q()));
        System.out.println(Arrays.toString(obj.v()));
        System.out.println(Arrays.toString(obj.a()));
        System.out.println(Arrays.toString(obj.j()));
        System.out.println(obj.pos(0.5) + "," + obj.vel(0.5) + "," + obj.acc(0.5) + "," + obj.jer(0.5));
    }
}
